tesp/package.py

```diff
@@ -787,3 +796,176 @@ def package(
 
         # finally the checksum
         create_checksum(out_path)
+
+
+def package_non_standard(outdir, granule):
+    """
+    yaml creator for the ard pipeline.
+    """
+
+    outdir = Path(outdir)
+    out_fname = Path(str(granule.wagl_hdf5).replace("wagl.h5", "yaml"))
+    out_fname_conv = Path(
+        str(granule.wagl_hdf5).replace("wagl.h5", "converted.datasets.h5")
+    )
+
+    f = h5py.File(outdir.joinpath(out_fname_conv))
+
+    with DatasetAssembler(metadata_path=out_fname, naming_conventions="dea") as da:
+        level1 = granule.source_level1_metadata
+        da.add_source_dataset(level1, auto_inherit_properties=True, inherit_geometry=True)
+        da.product_family = "ard"
+        da.producer = "ga.gov.au"
+        da.properties["odc:file_format"] = "HDF5"
+
+        with h5py.File(granule.wagl_hdf5, "r") as fid:
+            img_paths = [
+                ppjoin(fid.name, pth)
+                for pth in eodatasets3.wagl._find_h5_paths(fid, "IMAGE")
+            ]
+            granule_group = fid[granule.name]
+
+            try:
+                wagl_path, *ancil_paths = [
+                    pth
+                    for pth in (eodatasets3.wagl._find_h5_paths(granule_group, "SCALAR"))
+                    if "METADATA" in pth
+                ]
+            except ValueError:
+                raise ValueError("No nbar metadata found in granule")
+
+            [wagl_doc] = loads_yaml(granule_group[wagl_path][()])
+
+            da.processed = get_path(wagl_doc, ("system_information", "time_processed"))
+
+            org_collection_number = utils.get_collection_number(
+                da.producer, da.properties["landsat:collection_number"]
+            )
+
+            da.dataset_version = f"{org_collection_number}.1.0"
+            da.region_code = eodatasets3.wagl._extract_reference_code(da, granule.name)
+
+            eodatasets3.wagl._read_gqa_doc(da, granule.gqa_doc)
+            eodatasets3.wagl._read_fmask_doc(da, granule.fmask_doc)
+
+            with rasterio.open(granule.fmask_image) as ds:
+                fmask_layer = "/{}/OA_FMASK/oa_fmask".format(granule.name)
+                fmask_ds = f.create_dataset(
+                    fmask_layer, data=ds.read(1), compression='lzf', shuffle=True
+                )
+                fmask_ds.attrs["crs_wkt"] = ds.crs.wkt
+                fmask_ds.attrs["geotransform"] = ds.transform.to_gdal()
+
+                geotransform = ds.transform.to_gdal()
+
+                fmask_ds.attrs[
+                    "description"
+                ] = "Converted from ERDAS Imagine format to HDF5 "
+                    "to work with the limitations of varied formats within ODC"
+
+                grid_spec = images.GridSpec(
+                    shape=ds.shape,
+                    transform=Affine.from_gdal(*geotransform),
```

Comment:
Rather than having to convert to GDAL style and back to rasterio style, just pass it the `ds.transform`

Reply:
```suggestion
                    transform=ds.transform,
```

Reply:
this has been fixed